### ISSUE TITLE
Invalid TOTP URL for QR code

### ISSUE THREAD

**Issue by reepy** [jleclanche/python-bna#25]
I was trying to get `bna` to generate a URL, to create a QR code so I could add it to [AndOTP](https://github.com/andOTP/andOTP). However, the QR code wouldn't scan in AndOTP or Authy because of a missing semi colon. AndOTP says 'Invalid QR code' and Authy says 'Invalid format, token format not supported'.

As soon as the semi colon is added it works. The semi colon is present in the example in README.md under 'Getting an OTPAuth URL'.

I'm not sure if this is an issue in AndOTP or bna. Google Authenticator scans it without issue. The [spec](https://github.com/google/google-authenticator/wiki/Key-Uri-Format) doesn't specify it is required.

Steps to reproduce:

1. Setup `bna`
2. Run `bna show-url`
3. You'll notice the code is missing the colon before the question mark. This means you can't import this URL into common TOTP apps.

For example: `otpauth://totp/Blizzard:SERIAL?secret=SERCRET&issuer=Blizzard&digits=8`
Should be: `otpauth://totp/Blizzard:SERIAL:?secret=SERCRET&issuer=Blizzard&digits=8`

You can verify this:

1. Install a QR code generator `pip3 install qrcode[pil]`
2. Run `qr "PASTEURLHERE"`
3. Try and scan the QR code. You'll notice you get an error.
4. Repeat step 2, but add the missing semi colon.
5. Try and scan the QR code. You'll notice it works this time.

**Comment by DLslime** [comment]
No, it doesn't work. AndOTP still can't scan that QR code untill you delete `&digits=8` manually.

**Comment by mprobson** [comment]
Deleting the &digits=8 let me scan the code but then I was unable to use it (or modify it to generate more digits)

**Comment by DLslime** [comment]
AndOTP don't allow you to modify OTP's digits after you add it by scan the QR code. So anyway, it is definitely AndOTP's own defect. 
Other authentication Apps should be tested to see whether they also have the similar problem.

**Comment by reepy** [comment]
I tried other TOTP apps and it's in the original report (without removing `&digits=8`):

* AndOTP - 'Invalid QR code'
* Authy - 'Invalid format, token format not supported'
* FreeOTP - 'The token specified was invalid'.

I've just tried a few more:

* [PROTECTIMUS SMART OTP](https://play.google.com/store/apps/details?id=com.protectimus.android) - works
* [Sophos Authenticator](https://play.google.com/store/apps/details?id=com.sophos.sophtoken) - works
* [Microsoft Authenticator](https://play.google.com/store/apps/details?id=com.azure.authenticator) - works.

So maybe the standard is unclear and some are OK with the colon and others aren't?
